the-collab-lab/tcl-13-smart-shopping-list
# Rf rb add item form

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

In the Add Items Component / Route:
Created a form to add grocery list items to the database based on when the user plans to next purchase the item. 

In the View Items Component / Route:
Pulls grocery list items by user token and renders them as an unordered list to the component.

Stylistic Changes:
Created AddItems.css and Nav.css with styling outline to match the basic wireframe.

## Related Issue

closes #4

## Acceptance Criteria


Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”

Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![AddItems-Before](https://user-images.githubusercontent.com/10679592/96060337-130bba00-0e45-11eb-8831-cbc4bd78c306.JPG)
![ViewItems-Before](https://user-images.githubusercontent.com/10679592/96060345-19019b00-0e45-11eb-9fb4-87de2a6495b6.JPG)



### After

![AddItems-After](https://user-images.githubusercontent.com/10679592/96060355-1dc64f00-0e45-11eb-9db6-769051d0a3ee.JPG)
![ViewItems-After](https://user-images.githubusercontent.com/10679592/96060359-1f901280-0e45-11eb-80c9-695b28fa377e.JPG)



## Testing Steps / QA Criteria

In Add Items:
Timeframe should default to "Soon"
After clicking the "Add Item" button, an alert should confirm that the item was added to the database

In View List:
Items from the database should be rendered as a list

## TODO

Local storage user tokens should be linked to database read & write functions
